packages/go_router/test/parser_test.dart

```diff
@@ -100,6 +100,38 @@ void main() {
     expect(configuration.namedLocation('SNAKE_CASE'), '/hij?');
   });
 
+  test(
```

Comment:
Can you add a test in go_router_test.dart to test the entire flow from context.goNamed() to the actual page is built with the correct query params?

Reply:
Thanks for the suggestion, while writing the test, I actually faced an issue/bug:

If you run this code snippet:
```dart
final uri = Uri(path: 'a/b/', queryParameters: {'q1': 'v1', 'q2': ['v2', 'v3']});
print(uri.queryParametersAll);
print(uri.queryParameters);
```

It will log

```
{q1: [v1], q2: [v2, v3]}
{q1: v1, q2: v3}
```

Because of that, only carrying around `queryParams` is not enough, I had to add `queryParametersAll` everywhere.

Do you want me to remove `queryParams` as the data is now kind of duplicated? This would be a breaking change I think.
Or I could create getters `Map<String, dynamic> get queryParams` to not duplicate stored data and not introduce a breaking change?

What do you think?